src/Cake.Core/Graph/CakeGraph.cs

```diff
@@ -8,21 +8,37 @@
 
 namespace Cake.Core.Graph
 {
-    internal sealed class CakeGraph
+    /// <summary>
+    /// Represents the Cake task graph.
+    /// </summary>
+    public sealed class CakeGraph
```

Comment:
Made public due to the new tree functionality. Should we find another way of getting the task graph? Expose it by providing a read only interface to the tree script host somehow?